cai_causal_graph/time_series_causal_graph.py

```diff
@@ -1007,6 +1033,45 @@ def to_numpy_by_lag(self) -> Tuple[Dict[int, numpy.ndarray], List[str]]:
         assert self.variables is not None
         return adjacency_matrices, self.variables
 
+    def get_nodes_at_lag(self, time_lag: int = 0) -> List[TimeSeriesNode]:
+        """Return all nodes at time delta `time_lag`."""
+        return [node for node in self.get_nodes() if node.time_lag == time_lag]
+
+    def get_contemporaneous_nodes(self, node: NodeLike) -> List[TimeSeriesNode]:
+        """Return all nodes that are contemporaneous (i.e. have the same time_lag) to the provided node."""
+        assert node is not None, 'The `node` cannot be None.'
+        if isinstance(node, str):
+            node = self.get_node(node)
+
+        assert isinstance(node, TimeSeriesNode), 'The node must be a `TimeSeriesNode`.'
+        cont_nodes = self.get_nodes_at_lag(node.time_lag)
+        return [n for n in cont_nodes if n != node]
+
+    def get_potential_contemporanous_parents(self, node: NodeLike) -> List[TimeSeriesNode]:
+        """Return the true and potential parents that are contemporanous to the provided node."""
+        assert node is not None, 'The `node` cannot be None.'
+        if isinstance(node, str):
+            node = self.get_node(node)
+
+        assert isinstance(node, TimeSeriesNode), 'The node must be a `TimeSeriesNode`.'
+
+        contemporaneous_adj_nodes = []
+
+        # go through the inbounds edges
+        for edge in self.get_edges(destination=node):
+            assert isinstance(edge.source, TimeSeriesNode)   # for linting
+            if edge.source.time_lag == node.time_lag:
+                contemporaneous_adj_nodes.append(edge.source)
+
+        # go through the outbounds edges
+        for edge in self.get_edges(source=node):
+            if edge.edge_type == EdgeType.UNDIRECTED_EDGE:
```

Comment:
Why only undirected? Any of the unknown edges could be potential parents as well. Do you need the potential parents or just the true one? If only the latter, could confirm it is a DAG first. Assuming this will be useful for PCMCI, but not seeing a general use of this method. Currently, it won't really work if `EdgeType.UNKNOWN_DIRECTED` but that is a potential parent as well. May be better to not add this one to the graph class and just do in PCMCI. Can't see others using this often.